### PR TITLE
Export proposal struct data, add ChannelID

### PR DESCRIPTION
**Pull request by NiloCK**
Incremental toward #420

Two small changes in this PR:

1. the `Proposal` struct previously contained `toAdd`, `toRemove` - private fields which were not visible in other packages. (notably, the `protocols` packages, which want to act on this data). PR makes these fields public.
2. `Proposals` are generated by channels, and consumed by channels. As things stood, proposals went over-the-wire with no explicit labelling for the receiver. PR adds `ChannelID` to the proposal struct so that incoming proposals can easily be routed to the correct `left` or `right` ledgers.

There are some run-on changes to message and consensus_channel serialization, and also added tests.

---

#### Code quality

- [x] I have written clear commit messages
- [x] I have performed a self-review of my own code
- [x] This change does not have an unduly wide scope
- [x] I have separated logic changes from refactor changes (formatting, renames, etc.)
- [x] I have commented my code wherever necessary (can be 0)
- [x] I have added tests that prove my fix is effective or that my feature works, if necessary
- [x] I have added comprehensive [godoc](https://go.dev/blog/godoc) comments 

#### Project management

- [x] I have assigned myself to this PR
- [x] I have linked the appropriate github issue
- [x] I have assigned this PR to the appropriate GitHub project
- [x] I have assigned this PR to the appropriate GitHub Milestone

### CODE REVIEW

File: channel/state/outcome/guarantee.go [hunk 1]
```diff
@@ -7,8 +7,10 @@ import (
 
 // A Guarantee is an Allocation with AllocationType == GuaranteeAllocationType and Metadata = encode(GuaranteeMetaData)
 type GuaranteeMetadata struct {
-	Left  types.Destination // The peer who plays the role of Alice (peer 0)
-	Right types.Destination // The peer who plays the role of Bob (peer n+1, where n=len(intermediaries))
+	// The peer who plays the role of Alice (peer 0)
+	Left types.Destination
+	// The peer who plays the role of Bob (peer n+1, where n=len(intermediaries)
```
Comment (andrewgordstewart) [inline channel/state/outcome/guarantee.go:12]: This comment confuses me: in the virtual funding protocol, is there a guarantee that has `left=Alice` and `right=Bob`.

File: channel/consensus_channel/leader_channel_test.go [hunk 2]
```diff
@@ -197,3 +197,28 @@ func TestRestrictedFollowerMethods(t *testing.T) {
 		t.Errorf("Expected error when calling Receive as a leader, but found none")
 	}
 }
+
+func TestLeaderIncorrectlyAddressedProposals(t *testing.T) {
```
Comment (andrewgordstewart) [inline channel/consensus_channel/leader_channel_test.go:201]: On `main`, the leader_channel  tests have been updated. As follow-up work, this test could be written as
```
	{
		msg := "err:wrong channel"
		p := populatedQueue()[2]
		p4 := createAdd(p.TurnNum+10, types.Destination{11})
		p4.ChannelID = "wrong channel"
		counterP := bobSignedProposal(p.Vars, p4)
		t.Run(msg, testUpdateConsensusErr(counterP, ErrIncorrectChannelID))
	}
```

**Review by andrewgordstewart — APPROVED** [review]

File: channel/state/outcome/guarantee.go (hunk 1 above)
Comment (NiloCK) [reply channel/state/outcome/guarantee.go:12]: (This commit is accidentally part of this PR.)

Agreed that that comment is confusing, and in particular it does not reflect existing usage of this struct in the codebase. I had gotten confused on some overloading of leader / follower, left / right, etc, and intended to attach better godoc comments to existing structs.

File: channel/consensus_channel/leader_channel_test.go (hunk 2 above)
Comment (NiloCK) [reply channel/consensus_channel/leader_channel_test.go:201]: Noted!

**Pull request merged**